Source/Projections/Builder/ConventionProjectionBuilder.cs

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Dolittle.SDK.Events;
+using Dolittle.SDK.Events.Processing;
+using Dolittle.SDK.Projections.Internal;
+using Microsoft.Extensions.Logging;
+
+namespace Dolittle.SDK.Projections.Builder
+{
+    /// <summary>
+    /// Methods for building <see cref="IProjection{TReadModel}"/> instances by convention from an instantiated projection class.
+    /// </summary>
+    /// <typeparam name="TProjection">The <see cref="Type" /> of the read model.</typeparam>
+    public class ConventionProjectionBuilder<TProjection> : ICanBuildAndRegisterAProjection
+        where TProjection : class, new()
+    {
+        const string MethodName = "On";
+        readonly Type _projectionType = typeof(TProjection);
+
+        /// <inheritdoc/>
+        public void BuildAndRegister(
+            IEventProcessors eventProcessors,
+            IEventTypes eventTypes,
+            IEventProcessingConverter processingConverter,
+            ILoggerFactory loggerFactory,
+            CancellationToken cancellation)
+        {
+            var logger = loggerFactory.CreateLogger(GetType());
+            logger.LogDebug("Building projection from type {ProjectionType}", _projectionType);
+
+            if (!HasParameterlessConstructor())
+            {
+                logger.LogWarning("The projection class {ProjectionType} has no deafult/parameterless constructor", _projectionType);
+                return;
+            }
```

Comment:
Should we maybe log warnings as well if there are more than one constructor? Just in case someone tries IoC magic expecting it to work...